specs/schemes/exact/scheme_exact_algo.md

```diff
@@ -0,0 +1,222 @@
+- `lv`: The last valid round should be converted into a uint64 integer.
+
+#### Scheme
+
+Algorand uses [**msgpack**](https://www.npmjs.com/package/algorand-msgpack) to encode the transaction before signing.
+
+
+#### Steps to sign
+
+- Decode the `txn` parameters as described above.
+- Sort fields alphabetically. 
+- Encode the sorted fields using msgpack.
+- Add a prefix `TX` in its byte representation to the encoded message.
+- Sign the resulting byte array using the sender's private key with the `Ed25519` signature scheme.
+
+
+## Full `X-PAYMENT` header:
+
+```json
+{
+  "x402Version": 1,
+  "scheme": "exact",
+  "network": "algorand-mainnet",
+  "payload": {
+    "paymentTxID": "R4PFEBJGLXV4PDOIBLY7UKBLVLXFCPYLXGBILC22BAVILTKWV3TA", // ID of the transaction in the group that will pay the resource server
+    "paymentGroup": [
+      {
+        "signature": "VxpW6emWwct7ymzyXa/F2T7rwepM71XwsslPKEdpVqmbcRUB28QMdYJuuLkeP4dnVoI7emmxT6Q/YktqHc8xBg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "fee": "2000",
+          "fv": "6631154",
+          "lv": "6632154",
+          "snd": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "rcv": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "amnt": "0",
+          "type": "pay"
+        }
+      },
+      {
+        "signature": "wVGxbZwPd2c8zKX4RSE4muFRBOHe/nRq5VT6iMeMksi2agfOCTs+VzOGUPJ+bTT5T21EuYjoBKKDX2+/NZG8Cg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "asnd": "PPNHVQJGE6ZMEWPR35UHLUYMCCZV6VNTHLJMZDVCONXKUPV47K4WGQJWGI",
+          "arcv": "ACTSMNPJPS5JMZJJ5G73JOXUUMWXXDGS7TMOER3M4W7BC54ERSZ4KPSMHM",
+          "type": "axfer",
+          "fee": "0",
+          "fv": "6631154",
+          "lv": "6632154",
+          "xaid": "168103"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Verification
+
+Steps to verify a payment for the `exact` scheme on Algorand:
+
+1. Verify the signature is valid for each transaction in the `paymentGroup`.
+  a. Encode the `txn` parameters as described in the encoding section.
+  b. Recover the public key of the sender from the `snd` or `asnd` field by decoding from the address.
+  c. Verify the signature against the encoded message using the recovered public key.
+2. Verify that the atomic group of transactions is valid and can be executed as a single unit.
+3. Execute the payment group against an Algorand node's `simulate` endpoint to ensure the transactions would result in the amount specified in `paymentRequirements.maxAmountRequired` being transferred to the resource server's address.
+
+
+## Settlement
+
+Once the group is validated by the resource server, settlement can occur by the facilitator submitting the `paymentGroup` to the Algorand network through the `v2/transactions` endpoint against any valid Algorand node.
```

Comment:
It might be worth adding here that all that matters is verifying that the transaction with ID equal to `paymentTxID` is committed on the ledger (via Algod end-point).